React vs Web Components Documentation (#929)
department-of-veterans-affairs/vets-design-system-documentation: src/_about/for-developers.md

The majority of our web components also fire a `component-library-analytics` event used to translate component library actions into analytics data layer events. The event handler for this event exists in `vets-website`.

> ### Using React vs Web Components

Reviewer: Can this heading be worded to sound less competitive in an attempt to avoid any confusion? Maybe something like "React and Web Components"

While large portions of VA.gov are built via React applications there are some teams that cannot import React into their projects or have to perform work around hacks to have all React components available in order to be able to select the correct one to use.

> Due to these issues the Design System Team recommends using our Web Components on VA.gov applications and pages.

Reviewer 2: Could you add a second sentence here like "For easy identification, all of our Web Components begin with a `va-` prefix. For example, the Web Component version of our button component is named `va-button`." Or something to that effect.

Benefits include:

> - Future proofing as Web Components are agnostic and can be imported into any JS Framework

Reviewer 2: ```suggestion
- Future proofing as Web Components can be imported into any JS Framework
```

- Consistent syntax across frameworks and projects

> - Actively updated and maintained - we are deprecating most React components so they will not have the latest updates

Reviewer 2: ```suggestion
- Actively updated and maintained - we are deprecating most React components and they will not have the latest updates
```

> - Performance - no hacks needed to use Web Components directly on your project and you can select the specific ones you would like to use

Reviewer 2: I'm not sure what the second part of this sentence means.

> The Design System Team has specific linting and migration rules in place to help ease in the transition from React to Web Components and we encourage all developers to build out new components via Web Components moving forward on your projects.

Reviewer: Should we have a sentence up at the top of this section explicitly saying that teams are meant to use web components where appropriate, not write their own web components that they use in their app?

### React and Web Components
**Note:** Please use Web Components where applicable on projects. We maintain these components to provide VA teams with an ecosystem of vetted and tested components. These components are here to help scale and build projects quicker without needing to create net new components.

While large portions of VA.gov are built via React applications, there are some teams that cannot import React directly into their projects and have to add work around hacks in order to use React components.

Due to these issues the Design System Team recommends using our Web Components on VA.gov applications and pages.

For easy identification, all of our Web Components begin with a `va-` prefix. For example, the Web Component version of our alert component is named `va-alert`.

Benefits include:
- Future proofing as Web Components can be imported into any JS Framework
- Consistent syntax across frameworks and projects
- Actively updated and maintained - we are deprecating most React components and they will not have the latest updates
- Performance and speed

> The Design System Team has specific linting and migration rules in place to help ease in the transition from React to Web Components. We also encourage all developers to build out new components on applications via Design System Components instead of creating new similar bespoke components when building out an application.

Reviewer 2: ```suggestion
The Design System Team has specific linting and migration rules in place to help ease in the transition from React to Web Components. We also encourage all developers use Design System Components in their applications instead of creating their own similar components. If our components do not meet your needs, we would love to hear about it. Please reach out to us in Slack or [submit a bug report](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/new?assignees=caw310&labels=vsp-design-system-team&template=bug_template.md&title=). If you are interested in contributing a new component to the design system, please review our [documentation about that process](contributing-to-the-design-system).
```

I believe to make that last link work, you may have to add a "slug" to the YAML frontmatter on the contribution page.

Author: I tested and both links work correctly
